OutOfSchool/OutOfSchool.WebApi/Services/SocialGroupService.cs

```diff
@@ -44,23 +45,29 @@ public class SocialGroupService : ISocialGroupService
     }
 
     /// <inheritdoc/>
-    public async Task<IEnumerable<SocialGroupDto>> GetAll()
+    public async Task<IEnumerable<SocialGroupDto>> GetAll(LocalizationType localization = LocalizationType.Ua)
     {
-        logger.LogInformation("Getting all Social Groups started.");
+        logger.LogInformation($"Getting all Social Groups, {localization} localization, started.");
 
         var socialGroups = await repository.GetAll().ConfigureAwait(false);
-
+        var socialGroupsLocalized = socialGroups.Select(x =>
+        new SocialGroupDto()
+        {
+            Id = x.Id,
+            Name = localization == LocalizationType.En ? x.NameEn : x.Name,
+        });
         logger.LogInformation(!socialGroups.Any()
             ? "SocialGroup table is empty."
             : $"All {socialGroups.Count()} records were successfully received from the SocialGroup table");
 
-        return socialGroups.Select(socialGroup => mapper.Map<SocialGroupDto>(socialGroup)).ToList();
+        return mapper.Map<List<SocialGroupDto>>(socialGroupsLocalized);
```

Comment:
You can call method ToList when you doing selection. And after it use previeus result in logging and returning.